subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/InstallationSuppliers.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal;
+
+import java.io.File;
+import java.util.Collections;
+
+class InstallationSuppliers {
+
+    public static InstallationSupplier forDirectory(File directory) {
+        return () -> Collections.singleton(new InstallationLocation(directory, ""));
```

Comment:
I think it is a code smell that you pass an empty String in that API.
why not something like "from file"?

Reply:
Actually, I completely removed the suppliers factory. The only use in the meantime was in tests as all other supplier should not use this (but rather provide a proper source attribute).